app/controllers/users_controller.rb

```diff
@@ -22,8 +30,50 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.update_attributes user_params
```

Comment:
Will the update_attributes work if @user is nil? 
Can you try using "before_action" filter here. Study about filter in rails. 
Reference: [http://guides.rubyonrails.org/action_controller_overview.html#filters](http://guides.rubyonrails.org/action_controller_overview.html#filters)

Reply:
I add before_action :load_user, only: %i(show edit update destroy) check found user
    def load_user
      @user = User.find_by id: params[:id]
      return if @user
      flash[:danger] = t ".usernotfound"
      redirect_to root_path
    end